template.go

```diff
@@ -122,30 +145,41 @@ func RunTemplate(environment map[string]any, template Template) (string, error)
 }
 
 func goTemplate(template Template, environment map[string]any) (string, error) {
-	tpl := gotemplate.New("")
-
-	if template.LeftDelim != "" {
-		tpl = tpl.Delims(template.LeftDelim, template.RightDelim)
+	var tpl *gotemplate.Template
+	if len(template.Functions) == 0 {
+		// only use cache if there's no custom template functions because
```

Comment:
We can use the pointer value `reflect.ValueOf(f)` and append it to the template for a cache key ?